app/app.go

```diff
@@ -16,37 +17,41 @@ import (
 	"github.com/mattermost/mattermost-server/v5/services/httpservice"
 	"github.com/mattermost/mattermost-server/v5/services/imageproxy"
 	"github.com/mattermost/mattermost-server/v5/services/timezones"
+	"github.com/mattermost/mattermost-server/v5/store"
 	"github.com/mattermost/mattermost-server/v5/utils"
 )
 
 type App struct {
-	Srv *Server
-
-	Log              *mlog.Logger
-	NotificationsLog *mlog.Logger
-
-	T              goi18n.TranslateFunc
-	Session        model.Session
-	RequestId      string
-	IpAddress      string
-	Path           string
-	UserAgent      string
-	AcceptLanguage string
-
-	AccountMigration einterfaces.AccountMigrationInterface
-	Cluster          einterfaces.ClusterInterface
-	Compliance       einterfaces.ComplianceInterface
-	DataRetention    einterfaces.DataRetentionInterface
-	Elasticsearch    einterfaces.ElasticsearchInterface
-	Ldap             einterfaces.LdapInterface
-	MessageExport    einterfaces.MessageExportInterface
-	Metrics          einterfaces.MetricsInterface
-	Notification     einterfaces.NotificationInterface
-	Saml             einterfaces.SamlInterface
-
-	HTTPService httpservice.HTTPService
-	ImageProxy  *imageproxy.ImageProxy
-	Timezones   *timezones.Timezones
+	srv *Server
+
+	log              *mlog.Logger
+	notificationsLog *mlog.Logger
+
+	t              goi18n.TranslateFunc
+	session        model.Session
+	requestId      string
+	ipAddress      string
+	path           string
+	userAgent      string
+	acceptLanguage string
+
+	accountMigration einterfaces.AccountMigrationInterface
+	cluster          einterfaces.ClusterInterface
+	compliance       einterfaces.ComplianceInterface
+	dataRetention    einterfaces.DataRetentionInterface
+	elasticsearch    einterfaces.ElasticsearchInterface
+	ldap             einterfaces.LdapInterface
+	messageExport    einterfaces.MessageExportInterface
+	metrics          einterfaces.MetricsInterface
+	notification     einterfaces.NotificationInterface
+	saml             einterfaces.SamlInterface
+
+	httpService httpservice.HTTPService
+	imageProxy  *imageproxy.ImageProxy
+	timezones   *timezones.Timezones
+
+	context context.Context
+	store   store.Store
```

Comment:
Why adding `store` to the app struct instead of following the app -> server -> store path? I'm not 100% convinced on allowing for two paths to reach the same Store reference

Reply:
@mgdelacroix - I'll look into the possibility of keeping `store` in Server only.

Reply:
done. let's see if tests pass :)